sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -171,6 +172,11 @@ private[hive] class HiveClientImpl(
   private def newState(): SessionState = {
     val hiveConf = newHiveConf(sparkConf, hadoopConf, extraConfig, Some(initClassLoader))
     val state = new SessionState(hiveConf)
+    // When SessionState is initialized, the caller context is overridden by hive
+    // so we need to reset it back to the DRIVER
+    new CallerContext("DRIVER",
+      sparkConf.get(APP_CALLER_CONTEXT),
+      Option(sparkConf.getOption("spark.app.id").getOrElse(""))).setCurrentContext()
```

Comment:
similar to above, appId won't be absent here
```suggestion
      Some(sparkConf.getAppId)).setCurrentContext()
```

Reply:
Done.

Reply:
also `sparkConf.getOption("spark.app.id").getOrElse("")` => `sparkConf.getAppId`

Reply:
Done. Sorry I missed it earlier. Fixed it in the latest commit.